pkg/cosign/image_signer.go

```diff
@@ -1,34 +1,57 @@
 package cosign
 
 import (
+	"context"
 	"fmt"
 	"io/ioutil"
 	"log"
 	"os"
+	"time"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/pivotal/kpack/pkg/apis/build/v1alpha2"
+	cosignremote "github.com/sigstore/cosign/v2/pkg/oci/remote"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sclient "k8s.io/client-go/kubernetes"
 
 	"github.com/buildpacks/lifecycle/platform"
 	"github.com/pkg/errors"
 	"github.com/sigstore/cosign/v2/cmd/cosign/cli/options"
 )
 
-type SignFunc func(
-	ro *options.RootOptions, ko options.KeyOpts, signOpts options.SignOptions, imgs []string,
-) error
+type SignFunc func(*options.RootOptions, options.KeyOpts, options.SignOptions, []string) error
+
+type FetchSignatureFunc func(name.Reference, ...cosignremote.Option) (name.Tag, error)
+
+type BuilderSigner interface {
+	SignBuilder(context.Context, string, []corev1.Secret, authn.Keychain) ([]v1alpha2.SignaturePath, error)
+}
 
 type ImageSigner struct {
```

Comment:
Mayeb we could inject the k8s client here so that the (cluster)builder controller does not need to pass the k8s client through to the image signer

Reply:
To elaborate a little more, I am thinking that the FindCosignSecret method could be called by the RemoteBuilderCreater and the logic would all be contained there so that the builder reconcilers don't need to care about cosign. Just a thought and I don't really have a strong opinion on this

Reply:
I guess in that case you would have to pass in the service account to the remote builder create which probably does not make sense since the RemoteBuilderCreator doesn't deal with k8s resources like the reconcilers do